common/thorhelper/roxiehelper.cpp

```diff
@@ -2393,15 +2393,31 @@ void FlushingJsonBuffer::setScalarUInt(const char *resultName, unsigned sequence
 ClusterWriteHandler::ClusterWriteHandler(char const * _logicalName, char const * _activityType)
     : logicalName(_logicalName), activityType(_activityType)
 {
-    makePhysicalPartName(logicalName.get(), 1, 1, physicalName, false);
```

Comment:
Looks like you have moved this code from construction time to being done every time the information is asked for. Could that be a problem if caller asks a lot? Does it need caching?

Probably not, but just checking...

Reply:
I think it would mean that the function could potentially be called 3 times instead of once.
getLocalPhysicalFilename()
splitPhysicalFilename()
copyPhysical()
are each only called one time.

The function itself is not too expensive, and the change is required to support multiple storage planes properly.  So I think it is fine to no longer cache the results.

Reply:
ok